Kickstarter-iOS/Views/Controllers/CommentsViewControllerTests.swift

```diff
@@ -16,10 +16,47 @@ internal final class CommentsViewControllerTests: TestCase {
     super.tearDown()
   }
 
-  func testView() {
+  func testView_CurrentUser_LoggedOut() {
     Language.allLanguages.forEach { language in
-      withEnvironment(language: language) {
-        let controller = CommentsViewController()
+      withEnvironment(currentUser: nil, language: language) {
+        let controller = CommentsViewController.configuredWith(project: .template)
+        let (parent, _) = traitControllers(device: .phone4_7inch, orientation: .portrait, child: controller)
+        parent.view.frame.size.height = 1_100
+
+        self.scheduler.run()
+
+        controller.tableView.layoutIfNeeded()
+        controller.tableView.reloadData()
```

Comment:
Just for my own curiousity, is `layoutIfNeeeded` added as a extra layer of precaution to ensure the table view gets refreshed accurately. Just because I assume `reloadData()` also invalidates the layout.

Reply:
I copied the original template to write the subsequent tests and now realize it passes without those calls. Removed the lines. Good catch!

Reply:
Yes. Basically, I think it's saying update immediately even when reloadData() is called. This is useful because `layoutIfNeeded()` is a synchronous task.

Reply:
I ended up removing those two lines on all the tests because it's passing without it. I think `scheduler` handles manually advancing synchronous tasks and we're using it here.

Reply:
We've had to add these lines sometimes when snapshot tests keep failing, causing a layout pass and reloading has at times caused tests to work reliably that were otherwise flakey or kept failing on comparison. I think @moderateepheezy and I tried it here because he was seeing failures but if it's no longer needed that's good too 👍